src/main/java/com/exasol/adapter/installer/UserInputValidator.java

```diff
@@ -0,0 +1,101 @@
+package com.exasol.adapter.installer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * Validates user input.
+ */
+public class UserInputValidator {
+    private UserInputValidator() {
+    }
+
+    /**
+     * Validate an Exasol identifier.
+     *
+     * @param value the value
+     * @return validated string
+     */
+    public static String validateExasolIdentifier(final String value) {
+        final Set<Character> allowedSpecialCharacters = Set.of('_', '-', '.');
+        return validateString(value, allowedSpecialCharacters);
+    }
+
+    /**
+     * Validate a name of a file.
+     *
+     * @param value the value
+     * @return validated string
+     */
+    public static String validateFileName(final String value) {
+        final Set<Character> allowedSpecialCharacters = Set.of('_', '-', '.');
+        return validateString(value, allowedSpecialCharacters);
+    }
+
+    /**
+     * Validate a path.
+     *
+     * @param value the value
+     * @return validated string
+     */
+    public static String validatePath(final String value) {
```

Comment:
```suggestion
    public static String validatePath(final String path) {
```

Reply:
Renamed to keep consistent with other methods. Please check them and do not forget to change the Javadoc-s also.

Reply:
I think in the method validatePropertyKey it's not nice to have a parameter with "value" name... Maybe it's okay to keep them different?